modelz/aioclient.py

```diff
@@ -0,0 +1,137 @@
+from __future__ import annotations
+from typing import Any
+from http import HTTPStatus
+from urllib.parse import urljoin
+
+import aiohttp
+
+from rich.console import Console
+
+from .env import EnvConfig
+from .serde import Serde, SerdeEnum, TextSerde
+from modelz.utils import get_ssl_context_no_verify
+
+
+TIMEOUT = 300
+console = Console()
+DEFAULT_RESP_SERDE = TextSerde()
+DEFAULT_RETRY = 3
+
+
+class ModelzAuth:
+    def __init__(self, key: str | None = None) -> None:
+        config = EnvConfig()
+        self.key: str = key if key else config.api_key
+        if not self.key:
+            raise RuntimeError("cannot find the API key")
+
+    def get_headers(self) -> dict:
+        return {"X-API-Key": self.key}
+
+
+class ModelzResponse:
+    def __init__(self, resp: aiohttp.ClientResponse, serde: Serde = DEFAULT_RESP_SERDE):
+        """Modelz internal response."""
+        if resp.status != HTTPStatus.OK:
+            console.print(f"[bold red]err[{resp.status}][/bold red]: {resp.text}")
+            raise ValueError(f"inference err with code {resp.status}")
+        self.resp = resp
+        self.serde = serde
+        self._data = None
+
+    async def save_to_file(self, file: str):
+        with open(file, "wb") as f:
+            f.write(await self.data)
+
+    @property
+    async def data(self) -> Any:
+        if not self._data:
+            self._data = self.serde.decode(await self.resp.content.read())
+        return self._data
+
+    async def show(self):
+        console.print(await self.data)
+
+
+class ModelzClient:
+    def __init__(
+        self,
+        deployment: str | None = None,
+        key: str | None = None,
+        host: str | None = None,
+        timeout: float = TIMEOUT,
+    ) -> None:
+        # ...
+        config = EnvConfig()
+        self.host = host if host else config.host
+        self.deployment = deployment
+        self.auth = ModelzAuth(key)
+        self.timeout = timeout
+        self.session_request_kwargs = {}
+        if not getattr(config, "ssl_verify", True):
+            self.session_request_kwargs |= {"ssl": get_ssl_context_no_verify()}
```

Comment:
1. cannot get this key
2. use the normal way to add a key-value to the dict

Reply:
1. What do yoy mean by `cannot get this key`?

Reply:
2. resolved.

Reply:
1. I forgot to copy the update env.py to this PR😂 will fix